snorkel/map/core.py

```diff
@@ -106,9 +108,18 @@ class BaseMapper:
         Memoize mapper outputs?
     """
 
-    def __init__(self, name: str, pre: List["BaseMapper"], memoize: bool) -> None:
+    def __init__(
+        self,
+        name: str,
+        pre: List["BaseMapper"],
+        memoize: bool,
+        memoize_key: Optional[HashingFunction] = None,
```

Comment:
Any reason to prefer `None` as the default instead of just using `get_hashable` directly?

Reply:
I see 2 reasons :
- function are mutable and the "good practice" is to avoid mutable in default parameters. I don't really see a situation where we would mutate this function tho.
- we will have to import get_hashable in all the subclasses and all the wrappers of BaseMapper

Reply:
Great point about mutating functions in modules, hadn't thought of that route. Sounds good!